transport/graphsync/graphsync.go

```diff
@@ -375,10 +384,11 @@ func (t *Transport) CloseChannel(ctx context.Context, chid datatransfer.ChannelI
 		return nil
 	}
 	t.dataLock.Lock()
-	if _, ok := t.requestorCancelledMap[chid]; ok {
+	_, ok := t.requestorCancelledMap[chid]
+	t.dataLock.Unlock()
+	if ok {
```

Comment:
Woah. Great catch man ! This could have been causing deadlock issues ...

Reply:
Yeah I think it actually *has* caused some deadlock issues, I suspect once we release we'll see a bunch of weird problems go away.